### PR TITLE
Migrate to core 6.4.0

### PR DESCRIPTION
**Pull request by olperr1**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Dependency update

**What is the current behavior?**

powsybl-open-loadflow depends on powsybl-core 6.3.0.

**What is the new behavior (if this is a feature change)?**
powsybl-open-loadflow depends on powsybl-core 6.4.0.

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [X] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/util/Reports.java [hunk 1]
```diff
@@ -348,14 +348,9 @@ public static ReportNode createRootLfNetworkReportNode(int networkNumCc, int net
...
-                .withMessageTemplate("lfNetwork", "Network CC${networkNumCc} SC${networkNumSc}")
-                .withUntypedValue(NETWORK_NUM_CC, networkNumCc)
-                .withUntypedValue(NETWORK_NUM_SC, networkNumSc)
-                .add();
-        newReportNode.include(lfNetworkReportNode);
-        return newReportNode;
+    public static ReportNode includeLfNetworkReportNode(ReportNode reportNode, ReportNode lfNetworkReportNode) {
+        reportNode.include(lfNetworkReportNode);
+        return lfNetworkReportNode;
     }
```
Comment (vidaldid-rte) [inline src/main/java/com/powsybl/openloadflow/util/Reports.java:354]: @olperr1 why is this change part of an integration branch ? I didn't find the API change in core that requries to change this behaviour.

File: src/test/java/com/powsybl/openloadflow/sa/AbstractOpenSecurityAnalysisTest.java [hunk 2]
```diff
@@ -29,11 +29,7 @@
 import com.powsybl.openloadflow.network.LfBus;
 import com.powsybl.openloadflow.network.NodeBreakerNetworkFactory;
 import com.powsybl.openloadflow.network.SlackBusSelectionMode;
-import com.powsybl.security.LimitViolationFilter;
-import com.powsybl.security.SecurityAnalysisParameters;
-import com.powsybl.security.SecurityAnalysisReport;
-import com.powsybl.security.SecurityAnalysisResult;
-import com.powsybl.security.detectors.DefaultLimitViolationDetector;
+import com.powsybl.security.*;
 import com.powsybl.security.limitreduction.LimitReduction;
```
Comment (vidaldid-rte) [inline src/test/java/com/powsybl/openloadflow/sa/AbstractOpenSecurityAnalysisTest.java:33]: @olperr1 is this on purpose ? (just to be sure - I have no strong opinion about explicit or implicit import - but some projects prefer to always explicit the imports)

**Review by vidaldid-rte — COMMENTED** [review]
Just one question about a change in report that doesn't seem releated to integration with core. And one about collapsing import to be sure this was on purpose.

File: src/main/java/com/powsybl/openloadflow/util/Reports.java (hunk 1 above)
Comment (olperr1) [reply src/main/java/com/powsybl/openloadflow/util/Reports.java:354]: It's an adaptation for https://github.com/powsybl/powsybl-core/pull/3007.
When a `ReportNode` was included, its root was omitted and only its children were in fact included. The powsybl-core PR fixed that, so I reverted the workaround that was made in OLF.

File: src/test/java/com/powsybl/openloadflow/sa/AbstractOpenSecurityAnalysisTest.java (hunk 2 above)
Comment (olperr1) [reply src/test/java/com/powsybl/openloadflow/sa/AbstractOpenSecurityAnalysisTest.java:33]: It wasn't on purpose. I will revert this change.
Thanks.

**Review by vidaldid-rte — APPROVED** [review]
Thanks Olivier for your clarifications. All good for me.

File: src/main/java/com/powsybl/openloadflow/util/Reports.java (hunk 1 above)
Comment (jeandemanged) [reply src/main/java/com/powsybl/openloadflow/util/Reports.java:354]: Yes indeed at the time I made this code in #919 I went a bit fast and didn't fully realize it was actually a powsybl-core bug I was workaround-ing here ...

Only when I saw https://github.com/powsybl/powsybl-core/pull/3007 I understood this wasn't right and noted here: https://github.com/powsybl/powsybl-open-loadflow/pull/919#discussion_r1602876818

thanks @olperr1 for the fix

**Review by geofjamg — APPROVED** [review]

**Pull request merged**